### PR TITLE
auto enable interrupts

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you!

Thank you for your contribution.
Please make sure that your submission includes the following:

### Must

- [ ] The code compiles without `errors` or `warnings`.
- [ ] All examples work.
- [ ] `cargo fmt` was run.
- [ ] Your changes were added to the `CHANGELOG.md` in the proper section.
- [ ] You updated existing examples or added examples (if applicable).
- [ ] Added examples are checked in CI

Closes https://github.com/esp-rs/esp-hal/issues/908

### CODE REVIEW

**Comment by jessebraham** [comment]
I have an idea that might clean this up significantly, just need to see if it's possible first... Will report back soon.

**Comment by jessebraham** [comment]
Marking this as draft until CI is green, just to help me better keep track of things 😅

**Review by bjoernQ — APPROVED** [review]
Really nice - makes the examples look much cleaner and less scary 👍

File: esp-hal-common/src/embassy/mod.rs [hunk 1]
```diff
@@ -96,7 +98,70 @@ use time_driver::EmbassyTimer;
...
+    #[cfg(any(esp32))]
+    crate::interrupt::enable(Interrupt::SPI1_DMA, Priority::max()).unwrap();
+    #[cfg(any(esp32, esp32s2))]
+    crate::interrupt::enable(Interrupt::SPI2_DMA, Priority::max()).unwrap();
+    #[cfg(any(esp32, esp32s2))]
+    crate::interrupt::enable(Interrupt::SPI3_DMA, Priority::max()).unwrap();
+    #[cfg(esp32s2)]
+    crate::interrupt::enable(Interrupt::SPI4_DMA, Priority::max()).unwrap();
+
+    // see https://github.com/esp-rs/esp-pacs/issues/175
```
Comment (jessebraham) [inline esp-hal-common/src/embassy/mod.rs:138]: I've published new PACs for the ESP32-C6 and ESP32-H2, so this interrupt has now been renamed.

Comment (MabezDev) [reply esp-hal-common/src/embassy/mod.rs]: Thanks, this should just need a rebase once the pacs are updated in esp-hal.

**Review by jessebraham — APPROVED** [review]
LGTM, thanks! One less footgun 😁

**Pull request merged**